packages/circuits/circuits/filter.circom

```diff
@@ -0,0 +1,70 @@
+pragma circom 2.1.6; 
+
+include "circomlib/circuits/comparators.circom";
+
+// input selector and element
+// if selector[i] == 1 and element = i then return 1. 
+// We can optimize this template by precompute isOne array;
+template InSet() {
+    signal input selector[16];
+    signal input element; 
+    signal inside[16]; 
+    signal output out;
+    component isOne[16];
+    component equal[16];
+    for (var i = 0; i < 16; i++) {
+        isOne[i] = IsEqual();
+        isOne[i].in[0] <== selector[i];
+        isOne[i].in[1] <== 1;
+
+        equal[i] = IsEqual();
+        equal[i].in[0] <== element;
+        equal[i].in[1] <== i;
+
+        if (i > 0) {
+            inside[i] <== inside[i - 1] + isOne[i].out* equal[i].out;
+        } else {
+            inside[i] <== isOne[i].out * equal[i].out;
+        }
+    }
+
+    out <== inside[15];
+}
+
+template Filter(max_num_bytes) {
+    signal input data[max_num_bytes]; // private input;
+    signal input selector[16]; // public input;
+    signal output reveal_data[max_num_bytes];
```

Comment:
It would be great if we can optimize the size of this array - the output will have this size even if only few values are revealed. This will increase the calldata for on-chain calls to the verifier.

Maybe we can shift non-zero values to one side with a delimiter in between field values? The contract can take only the data bits and then pad with zeros internally before verifying? 🤔 
Anyway - a thought for future PR